pkg/objstore/s3/s3.go

```diff
@@ -134,6 +138,17 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 				},
 			},
 		}
+		creds = credentials.NewChainCredentials(chain)
+
+		// We will watch for credential changes for non-static credentials
+		filename := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
```

Comment:
We should probably also get filename from config if we can. Then provide that to FileAWSCredentials. I really hate having to duplicate the FileAWSCredentials logic to determine filename here.